core/drivers/scmi-msg/voltage_domain.c

```diff
@@ -0,0 +1,416 @@
+	scmi_status_response(msg, status);
+}
+
+static void scmi_voltd_config_get(struct scmi_msg *msg)
+{
+	const struct scmi_voltd_config_get_a2p *in_args = (void *)msg->in;
+	struct scmi_voltd_config_get_p2a out_args = { };
+	unsigned int domain_id = 0;
+
+	if (msg->in_size != sizeof(*in_args)) {
+		scmi_status_response(msg, SCMI_PROTOCOL_ERROR);
+		return;
+	}
+
+	if (in_args->domain_id >= plat_scmi_voltd_count(msg->agent_id)) {
+		scmi_status_response(msg, SCMI_INVALID_PARAMETERS);
+		return;
+	}
+
+	domain_id = confine_array_index(in_args->domain_id,
+					plat_scmi_voltd_count(msg->agent_id));
+
+	if (plat_scmi_voltd_get_config(msg->agent_id, domain_id,
+				       &out_args.config)) {
+		scmi_status_response(msg, SCMI_INVALID_PARAMETERS);
+		return;
+	}
+
+	scmi_write_response(msg, &out_args, sizeof(out_args));
+}
+
+static void scmi_voltd_level_set(struct scmi_msg *msg)
+{
+	const struct scmi_voltd_level_set_a2p *in_args = (void *)msg->in;
+	int32_t status = SCMI_GENERIC_ERROR;
+	unsigned int domain_id = 0;
+
+	if (msg->in_size != sizeof(*in_args)) {
+		scmi_status_response(msg, SCMI_PROTOCOL_ERROR);
+		return;
+	}
+
+	if (in_args->domain_id >= plat_scmi_voltd_count(msg->agent_id)) {
+		scmi_status_response(msg, SCMI_INVALID_PARAMETERS);
+		return;
+	}
+
+	domain_id = confine_array_index(in_args->domain_id,
+					plat_scmi_voltd_count(msg->agent_id));
+
+	status = plat_scmi_voltd_set_level(msg->agent_id, domain_id,
+					   in_args->voltage_level);
+
+	scmi_status_response(msg, status);
+}
+
+static void scmi_voltd_level_get(struct scmi_msg *msg)
+{
+	const struct scmi_voltd_level_get_a2p *in_args = (void *)msg->in;
+	struct scmi_voltd_level_get_p2a out_args = {
+		.status = SCMI_SUCCESS,
+	};
+	unsigned int domain_id = 0;
+
+	if (msg->in_size != sizeof(*in_args)) {
+		scmi_status_response(msg, SCMI_PROTOCOL_ERROR);
+		return;
+	}
+
+	if (in_args->domain_id >= plat_scmi_voltd_count(msg->agent_id)) {
+		scmi_status_response(msg, SCMI_INVALID_PARAMETERS);
+		return;
+	}
+
+	domain_id = confine_array_index(in_args->domain_id,
+					plat_scmi_voltd_count(msg->agent_id));
+
+	out_args.voltage_level = plat_scmi_voltd_get_level(msg->agent_id,
+							   domain_id);
+
+	scmi_write_response(msg, &out_args, sizeof(out_args));
+}
+
+static const scmi_msg_handler_t scmi_voltd_handler_table[] = {
+	[SCMI_PROTOCOL_VERSION] = report_version,
+	[SCMI_PROTOCOL_ATTRIBUTES] = report_attributes,
+	[SCMI_PROTOCOL_MESSAGE_ATTRIBUTES] = report_message_attributes,
+	[SCMI_VOLTAGE_DOMAIN_ATTRIBUTES] = scmi_voltd_domain_attributes,
+	[SCMI_VOLTAGE_DESCRIBE_LEVELS] = scmi_voltd_describe_levels,
+	[SCMI_VOLTAGE_CONFIG_SET] = scmi_voltd_config_set,
+	[SCMI_VOLTAGE_CONFIG_GET] = scmi_voltd_config_get,
+	[SCMI_VOLTAGE_LEVEL_SET] = scmi_voltd_level_set,
+	[SCMI_VOLTAGE_LEVEL_GET] = scmi_voltd_level_get,
+};
+
+static bool message_id_is_supported(size_t message_id)
+{
+	return message_id < ARRAY_SIZE(scmi_voltd_handler_table) &&
+	       scmi_voltd_handler_table[message_id];
```

Comment:
Personally I would use shorter identifiers. Long names for file scope variables make little sense IMO.
```C
        return id < ARRAY_SIZE(handlers) && handlers[id];
```

Reply:
true. I'll shorten it.

Reply:
done